### PR TITLE
Keypad

### PR DESCRIPTION
**Pull request by fitzsimonsdevin**
# Summary of changes
Begin implementing Keypad
- Several updates to component state to align more closely with project tutorial
- Test, proptypes and passing updates to Keypad

# Acceptance Criteria
- [x] My code follows the style guidelines of this project
    * ~**Including** IE11 Support~
    * **Including** 1024 min. width
- [ ] I have run eslint and prettier and have no errors in my files
- [x] I have performed a self-review of my own code
- [ ] CodeClimate has no errors. If there are CodeClimate errors that you choose not to fix, please note why below and initiate a discussion with the team concerning the validity of the errors.

### CODE REVIEW

File: src/components/Keypad/Keypad.spec.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+                numbers={[]}
+                operators={[]}
+                setOperator={jest.fn()}
+                updateDisplay={jest.fn()}
+            />
+        );
+    });
+
+    it('should render a <div />', () => {
+        expect(wrapper.find('div').length).toEqual(1);
```
Comment (jacobvenable) [inline src/components/Keypad/Keypad.spec.js]:
```suggestion
        expect(wrapper.find('div').length).toBe(1);
```
Just to be strict like Miss Honey.

File: src/components/Keypad/Keypad.spec.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+                callOperator={jest.fn()}
+                numbers={[]}
+                operators={[]}
+                setOperator={jest.fn()}
+                updateDisplay={jest.fn()}
+            />
+        );
+    });
+
+    it('should render a <div />', () => {
```
Comment (jacobvenable) [inline src/components/Keypad/Keypad.spec.js]:
```suggestion
    it('renders a <div />', () => {
```
What it _is_ instead of what it _should be_.

File: src/components/Keypad/Keypad.spec.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import Keypad from './Keypad';
+
+describe('Keypad', () => {
+    let wrapper;
+    beforeEach(() => {
+        wrapper = shallow(
+            <Keypad
+                callOperator={jest.fn()}
```
Comment (jacobvenable) [inline src/components/Keypad/Keypad.spec.js]: Ok for now, but maybe in a refactor of the test you could spread props like before.

File: src/components/Keypad/Keypad.spec.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import Keypad from './Keypad';
+
+describe('Keypad', () => {
+    let wrapper;
```
Comment (jacobvenable) [inline src/components/Keypad/Keypad.spec.js]:
```suggestion
    let subject;
```
Just a convention update.

File: src/components/Keypad/Keypad.jsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    
+    </div>
+)
+
+Keypad.propTypes = {
+    callOperator: func.isRequired,
+    numbers: array.isRequired,
+    operators: array.isRequired,
+    setOperator: func.isRequired,
+    updateDisplay: func.isRequired,
```
Comment (jacobvenable) [inline src/components/Keypad/Keypad.jsx]: I know you're following a tutorial, but just a note: it's best to add these props as you start implementing them.

Once you're ready to build out the things with the `callOperator()`, then you add the prop. But ya, I understand it's probz the tutorial 👍

Comment (fitzsimonsdevin) [reply src/components/Keypad/Keypad.jsx]: Yeah If I wasn't moving on to part two on another url I'd have continued holding off. I did go through the exercise of doing them one at a time for the sake of building habits. But ya... #tutorialz4Lyfe

File: src/components/Calculator/Calculator.spec.js [hunk 6]
```diff
@@ -17,4 +18,19 @@ describe('Calculator', () => {
             .prop('displayValue'))
         .toBe(subject.instance().state.displayValue)
     })
+
+    it ('contains a Keypad and Display component', () => {
+        expect(subject
+            .containsAllMatchingElements([
+                <Display displayValue={subject.instance().state.displayValue} />,
+                <Keypad
+                    callOperator={subject.instance().callOperator}
```
Comment (jacobvenable) [inline src/components/Calculator/Calculator.spec.js]: Maybe consider testing each of the `<Keypad/>` expected props individually rather than all together. Similar to this [comment](https://github.com/fitzsimonsdevin/react-calculator/pull/4#discussion_r290525107).

File: src/components/Calculator/Calculator.spec.js [hunk 7]
```diff
@@ -17,4 +18,19 @@ describe('Calculator', () => {
             .prop('displayValue'))
         .toBe(subject.instance().state.displayValue)
     })
+
+    it ('contains a Keypad and Display component', () => {
+        expect(subject
+            .containsAllMatchingElements([
+                <Display displayValue={subject.instance().state.displayValue} />,
```
Comment (jacobvenable) [inline src/components/Calculator/Calculator.spec.js]: Test for the rendering of `<Display/>` and `<Keypad/>` separately.

File: src/components/Keypad/Keypad.jsx (hunk 5 above)
Comment (jacobvenable) [reply src/components/Keypad/Keypad.jsx]: yolo

File: src/components/Calculator/Calculator.spec.js [hunk 8]
```diff
@@ -1,7 +1,8 @@
 import React from 'react';
 import { shallow } from 'enzyme';
 import Calculator from './Calculator';
-import Display from '../Display/Display';
+import Display from '../Display';
+import Keypad from '../Keypad';
```
Comment (jacobvenable) [inline src/components/Calculator/Calculator.spec.js]: Since we're still sticking to unit tests, I'd go ahead and mock both `<Display/>` and `<Keypad/>`

File: src/.nvmrc [hunk 9]
```diff
@@ -0,0 +1 @@
+v8.15.0
```
Comment (jacobvenable) [inline src/.nvmrc]: NBD, but if I saw this in EXH, I'd probably ask that it be in a separate PR unless it was relevant to the changes. No need to do anything for this though.

File: src/components/Calculator/Calculator.spec.js (hunk 6 above)
Comment (fitzsimonsdevin) [reply src/components/Calculator/Calculator.spec.js]: would you do roughly:
```
    it ('contains a Keypad componet', () => {
        const keypad = subject.find('Keypad')
        
        expect(keypad
            .prop('numbers'))
        .toBe(subject.instance().state.numbers)
        
        expect(keypad
            .prop('operators'))
        .toBe(subject.instance().state.operators)
    })
```

File: src/.nvmrc (hunk 9 above)
Comment (kyleruschel) [reply src/.nvmrc]: This was me, it's not part of the tutorial but I wanted it in here so we were all using the correct version of node

File: src/components/Calculator/Calculator.spec.js (hunk 6 above)
Comment (jacobvenable) [reply src/components/Calculator/Calculator.spec.js]: Ya, along those lines. If you wanted to make it more concise, you could create an individual variable:
```js
//this in the beforeEach()  (or wherever subject is declared)
const numbersState = subject.instance().state.numbers;
...

//this in the assertion
expect(keypad.prop('numbers')).toBe(numbersState)
```
